packages/core/src/components/RadioButton/RadioButton.tsx

```diff
@@ -52,13 +54,14 @@ const RadioButton: React.FC<RadioButtonProps> = ({
     }
   };
 
-  const selected = contextValue === value;
+  const isSelected =
+    selected || (contextValue != null && contextValue === value);
```

Comment:
RadioButton is utilized by RadioButtonRow, which passes down a `selected` prop. RadioButtons can also be used standalone inside a RadioButtonGroup, hence the need for also reading `contextValue`.